apps/omg_api/lib/state/transaction.ex

```diff
@@ -223,4 +219,31 @@ defmodule OMG.API.State.Transaction do
   """
   @spec get_outputs(t()) :: list(output())
   def get_outputs(%__MODULE__{outputs: outputs}), do: outputs
+
+  defp inputs_without_gaps([input | _] = inputs) do
```

Comment:
It looks like `init` can be chipped off in `check_for_gaps`, I think it would be a bit more readable thanks to that.

Next and more general: is there a good reason why `inputs_without_gaps` is so different from `outputs_...`? I'd be great if they only differed in their `check_for_gasp` invocation.

Even nexter: hey, have you considered doing `Enum.chunk` to slice the `inputs`/`outputs` into all `{n, n+1}` pairs and see if there is no `{empty, nonempty}` pair? I think this would spare us the `reduce` and be very readable